packages/fxa-settings/src/pages/Signin/utils.ts

```diff
@@ -58,7 +58,14 @@ export async function handleNavigation(
     return { error };
   }
 
-  if (tempHandleSyncLogin && navigationOptions.integration.isSync()) {
+  if (
+    tempHandleSyncLogin &&
+    navigationOptions.integration.isSync() &&
+    // If the _next page_ is `signin_totp_code`, we also don't want to send this
+    // because we end up sending it twice with the first message containing
+    // `verified: false`, causing a Sync sign-in issue (see FXA-9837).
+    !to?.includes('signin_totp_code')
```

Comment:
See comment above this function too, we'll be refactoring this stuff once we convert `connect_another_device`.